vpr/src/analytical_place/global_placer.cpp

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file
+ * @author  Alex Singer
+ * @date    October 2024
+ * @brief   The definitions of the global placers used in the AP flow and their
+ *          base class.
+ */
+
+#include "global_placer.h"
+#include <cstdio>
+#include <memory>
+#include "analytical_solver.h"
+#include "ap_netlist.h"
+#include "partial_legalizer.h"
+#include "partial_placement.h"
+#include "vpr_error.h"
+#include "vtr_log.h"
+#include "vtr_time.h"
+
+std::unique_ptr<GlobalPlacer> make_global_placer(e_global_placer placer_type,
+                                                 const APNetlist& netlist) {
+    // Based on the placer type passed in, build the global placer.
+    switch (placer_type) {
+        case e_global_placer::SimPL:
+            return std::make_unique<SimPLGlobalPlacer>(netlist);
+        default:
+            VPR_FATAL_ERROR(VPR_ERROR_AP,
+                            "Unrecognized global placer type");
+
+    }
+}
+
+SimPLGlobalPlacer::SimPLGlobalPlacer(const APNetlist& netlist) : GlobalPlacer(netlist) {
+    // This can be a long method. Good to time this to see how long it takes to
+    // construct the global placer.
+    vtr::ScopedStartFinishTimer global_placer_building_timer("Constructing Global Placer");
+    // Build the solver.
+    solver_ = make_analytical_solver(e_analytical_solver::QP_HYBRID,
+                                     netlist);
+    // Build the partial legalizer
+    partial_legalizer_ = make_partial_legalizer(e_partial_legalizer::FLOW_BASED,
+                                                netlist);
+}
+
+/**
+ * @brief Helper method to print the header of the per-iteration status updates
+ *        of the global placer.
+ */
+static void print_SimPL_status_header() {
+    VTR_LOG("---- ---------------- ---------------- ----------- -------------- ----------\n");
+    VTR_LOG("Iter Lower Bound HPWL Upper Bound HPWL Solver Time Legalizer Time Total Time\n");
```

Comment:
Might want to space out the header more, as it will be harder to see how to read the spacing within a phrase and across phrases is the same (Iter Lower Bound    HPWL vs. Iter    Lower Bound HPWL). although the dashes below will help.  You can shorten some wording if that makes it too long, e.g.
Iter   L. Bound HPWL ...